### DISCUSSION TITLE
Flutter app doesn't work more than once

### DISCUSSION THREAD

**Discussion by NicoNick-2** [community/community#131418] [Programming Help]
### Body

I set up the emulator using Android Studio. Then I created a flutter project using "flutter create first_try" and opened it in VS Code. When I run the program for the first time (When you create a new project, it has an example code to execute in main.dart) it works fine but after I stop it and run it again it doesn't show some features. I don't know what this is related to but if I wipe emulator's data it works fine again; however closing and reopening it gives me problems again.

It doesn't give me any errors, anything; it just doesn't display things such as texts or buttons.

This is the code, I didn't touch it:

```import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // TRY THIS: Try running your application with "flutter run". You'll see
        // the application has a purple toolbar. Then, without quitting the app,
        // try changing the seedColor in the colorScheme below to Colors.green
        // and then invoke "hot reload" (save your changes or press the "hot
        // reload" button in a Flutter-supported IDE, or press "r" if you used
        // the command line to start the app).
        //
        // Notice that the counter didn't reset back to zero; the application
        // state is not lost during the reload. To reset the state, use hot
        // restart instead.
        //
        // This works for code too, not just values: Most code changes can be
        // tested with just a hot reload.
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // TRY THIS: Try changing the color here to a specific color (to
        // Colors.amber, perhaps?) and trigger a hot reload to see the AppBar
        // change color while the other colors stay the same.
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          //
          // TRY THIS: Invoke "debug painting" (choose the "Toggle Debug Paint"
          // action in the IDE, or press "p" in the console), to see the
          // wireframe for each widget.
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```

This is what is displayed with the command "flutter doctor -v":

  [√] Flutter (Channel stable, 3.22.2, on Microsoft Windows [Versione 10.0.19045.4529], locale it-IT)
      • Flutter version 3.22.2 on channel stable at C:\Flutter\flutter
      • Upstream repository https://github.com/flutter/flutter.git
      • Framework revision 761747bfc5 (5 weeks ago), 2024-06-05 22:15:13 +0200
      • Engine revision edd8546116
      • Dart version 3.4.3
      • DevTools version 2.34.3
  
  [√] Windows Version (Installed version of Windows is version 10 or higher)      
  
  [√] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
      • Android SDK at C:\Users\nicol\AppData\Local\Android\sdk
      • Platform android-35, build-tools 35.0.0
      • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java      
      • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)      
      • All Android licenses accepted.
  
  [√] Chrome - develop for the web
      • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe
  
  [!] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.3.1)
      • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
      • Visual Studio Community 2022 version 17.3.32811.315
      X Visual Studio is missing necessary components. Please re-run the Visual Studio installer for the
        "Desktop development with C++" workload, and include these components:
          MSVC v142 - VS 2019 C++ x64/x86 build tools
           - If there are multiple build tool versions available, install the latest
          C++ CMake tools for Windows
          Windows 10 SDK
  
  [√] Android Studio (version 2024.1)
      • Android Studio at C:\Program Files\Android\Android Studio
      • Flutter plugin can be installed from:
         https://plugins.jetbrains.com/plugin/9212-flutter
      • Dart plugin can be installed from:
         https://plugins.jetbrains.com/plugin/6351-dart
      • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)
  
  [√] VS Code (version 1.91.0)
      • VS Code at C:\Users\nicol\AppData\Local\Programs\Microsoft VS Code
      • Flutter extension version 3.92.0
  
  [√] Connected device (4 available)
      • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 15 (API 35) (emulator)    
      • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Versione       
        10.0.19045.4529]
      • Chrome (web)                 • chrome        • web-javascript • Google Chrome 126.0.6478.127      
      • Edge (web)                   • edge          • web-javascript • Microsoft Edge 126.0.2592.87      

  [√] Network resources
      • All expected network resources are available.
  
  ! Doctor found issues in 1 category.

![NoWorks](https://github.com/community/community/assets/134071292/e6c178c5-1edc-4b5f-b1ec-27559a5ab898)
![YesWorks](https://github.com/community/community/assets/134071292/d7c34fac-e025-457b-a92e-53ec3a3459ab)

Thanks to anyone who is willing to help me :)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by NicoNick-2** [comment]
Doing some tests I have noticed that switching between modes makes the app working again. If I start it using "flutter run" it works but only once, when I start it using "flutter run --release" it creates a release and launches it; now it works again. However it works only once per mode if I can say so, it seems like I have to swap between modes everytime in order to make it work all times.

**Comment by NicoNick-2** [comment]
Now I connected my personal phone to the computer and installed the app via USB, everything works perfectly and there are no issues, therefore the problem is concerned with the emulator.

**Reply by DebrayLeo** [reply]
I had several problems with my emulator on VS Code too, I just switched to Android Studio with the same emulator and the same code and it works well.

**Comment by odelyo** [comment]
1. Check for Emulator Settings and Updates:
Ensure that your Android emulator is fully updated. Sometimes issues like these can be related to the emulator itself rather than your Flutter code.
Try different emulator configurations (e.g., switching between different API levels) to see if the issue persists across different setups.
2. Check for Cached Data Issues:
If wiping the emulator's data resolves the issue temporarily, it might indicate a problem with how the app or emulator handles cached data. You can try the following:
Run flutter clean in your project directory to clear the build cache, and then run the app again.
Delete the app from the emulator manually and reinstall it by running the project again.
3. Hot Restart vs. Hot Reload:
Ensure that you're using "Hot Restart" instead of "Hot Reload" after making significant changes to the app or restarting it. "Hot Reload" might not fully reset the app's state, which could lead to issues like the ones you're describing.
4. Check for Emulator Logs:
Open the Android Studio logcat window and check the logs when you run the app the second time. Look for any errors or warnings that might give clues about what's going wrong.
5. Test on a Physical Device:
If possible, try running your Flutter app on a physical Android device. This will help you determine if the issue is specific to the emulator.
6. Check for Flutter Version Issues:
Sometimes, issues can arise from bugs in specific Flutter versions. Ensure that your Flutter installation is up to date. You can update Flutter using the command:
`flutter upgrade`
7. Check the ThemeData:
While it's unlikely, the issue could be related to how the theme is applied. Try removing or modifying the colorScheme in your ThemeData to see if it affects the problem.
8. Check the Visual Studio Configuration (Windows-specific):
You mentioned that the "Flutter Doctor" command indicated an issue with your Visual Studio setup. While this is more related to Windows app development, ensuring that your Visual Studio installation is correctly configured might help prevent any unexpected interactions.
9. Reinstall the Emulator:
As a last resort, try uninstalling and reinstalling the Android emulator or creating a new emulator from scratch. This can sometimes resolve persistent issues that aren't related to your code.
